porky_lib.gemspec

```diff
@@ -27,24 +27,29 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'byebug'
   spec.add_development_dependency 'codacy-coverage'
   spec.add_development_dependency 'codecov'
+  spec.add_development_dependency 'factory_bot_rails'
   spec.add_development_dependency 'msgpack'
   spec.add_development_dependency 'rake'
   spec.add_development_dependency 'rbnacl', '=5.0.0'
   spec.add_development_dependency 'rbnacl-libsodium'
   spec.add_development_dependency 'rspec'
   spec.add_development_dependency 'rspec-collection_matchers'
   spec.add_development_dependency 'rspec-mocks'
+  spec.add_development_dependency 'rspec-rails'
   spec.add_development_dependency 'rspec_junit_formatter'
   spec.add_development_dependency 'rubocop'
   spec.add_development_dependency 'rubocop-performance'
   spec.add_development_dependency 'rubocop-rspec'
   spec.add_development_dependency 'rubocop_runner'
   spec.add_development_dependency 'simplecov'
+  spec.add_development_dependency "sqlite3"
```

Comment:
why sqlite3 here?

Reply:
sqlite3 is used here since it support a database to be temporary (stored in memory) for testing, since PorkyLib doesn't have a database

Reply:
from what I've seen, postgresql doesn't support that

Reply:
it's not clear to me why PorkyLib needs its own database. What is it storing here? Shouldn't data storage be the responsibility of the app using PorkyLib?

Reply:
Yep, the apps using PorkyLib will be providing their database, and will be using the `has_encrypted` and `partition_provider` concern, but to test if the concerns are doing the right thing, I need to use some models, then some factories for those models, which requires a database. But if I do what @bcarr092 suggested above, I will not need factories, and maybe also no database

Reply:
the database is only required to test the concerns. can you use a memory database?

Reply:
That is what I'm doing in the `spec_helper`, is it what you are talking about memory database?
```ruby
ActiveRecord::Base.establish_connection adapter: "sqlite3", database: ":memory:"
```

Reply:
From what I gather from the above, this is working as intended.  The database is only required for test environment (only loaded in `spec_helper`) and is indeed set to be used as an in-memory database.  From https://stackoverflow.com/questions/7872693/running-postgresql-in-memory-only it does appear this is not possible with PostgreSQL.